File: 2452. Words Within Two Edits of Dictionary/twoEditWords.java
class Solution {
    public List<String> twoEditWords(String[] queries, String[] dictionary) {
        
        List<String> list= new ArrayList<>();

        for(String q: queries){
            for(String d: dictionary){
                if(isValid(q,d)){
                    // System.out.println(isValid(q,d));
                    list.add(q);
                    break; //to avoid adinf more word from dictionary coz we got a match for q
                }
            }
        }
        return list;
        
    }

    private boolean isValid(String q, String d){
        int k=0;
        for(int i=0; i<q.length(); i++){
        
            if(q.charAt(i)!=d.charAt(i)){
                k++;
                if(k>2){
                    return false;
                 
                }
            }   

        
        }
        return true;
    }
}

File: 2452. Words Within Two Edits of Dictionary/twoEditWords.py
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        res =[]
        for q in queries:
            for d in dictionary:
                if self.isValid(q,d):
                    res.append(q)
                    break
        return res
    

    def isValid(self, s, d):
        k=0
        for i in range(len(s)):
            if(s[i]!=d[i]):
                k+=1
                if(k>2):
                    return False
        return True
